src/App.tsx

```diff
@@ -1,17 +1,122 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import React, {
+  useContext, useEffect, useRef, useState,
+} from 'react';
+
+import { getTodos, postTodos, deleteTodos } from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { Filter } from './components/React/Filter';
+import { NewTodo } from './components/React/NewTodo';
+import { TodoList } from './components/React/TodoList';
+import { Todo } from './types/Todo';
+
+enum Errors {
+  Server = 'Unable to load data from the Server',
+  Post = 'Unable to add the Todo',
+  Add = 'Title can\'t be empty',
+  Delete = 'Unable to delete a todo',
+  Update = 'Unable to update a todo',
+}
 
 export const App: React.FC = () => {
   // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
-  useEffect(() => {
-    // focus the element with `ref={newTodoField}`
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [input, setInput] = useState('');
+  const [isAdding, setIsAdding] = useState(false);
+  const [todoOnload, setTodoOnLoad] = useState<Todo | null>(null);
+  const [todoIdsOnRemove, setTodoIdsOnRemove] = useState<number[]>([]);
+  const [isCompleted] = useState(false);
+
+  const focusInput = () => {
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
+  };
+
+  const downloadDataAPI = () => {
+    if (user) {
+      getTodos(user.id)
+        .then(todosFromServer => {
+          setTodos(todosFromServer);
+          setTodoOnLoad(null);
+          focusInput();
+        })
+        .catch(() => setError(Errors.Server));
+    }
+  };
+
+  const deleteDataAPI = (id: number) => {
+    deleteTodos(id)
+      .then(() => downloadDataAPI())
+      .catch(() => {
+        setError(Errors.Delete);
+        setTodoIdsOnRemove([]);
+      });
+  };
+
+  const handleTodoDeleteButton = (todoId: number) => {
+    deleteDataAPI(todoId);
+    setTodoIdsOnRemove([...todoIdsOnRemove, todoId]);
+  };
+
+  const handleClearButton = () => {
+    todos
+      .filter(todo => todo.completed)
```

Comment:
you should use `setTodos`. it won't re-render data

Reply:
<img width="351" alt="image" src="https://user-images.githubusercontent.com/104283893/214373749-33c9c282-7846-45fe-a1bc-c0dd7340fe8e.png">